topdown/trusted_identity_providers.go

```diff
@@ -0,0 +1,121 @@
+package topdown
+
+import (
+	"errors"
+	"fmt"
+	"github.com/coreos/go-oidc"
+	jwt "github.com/square/go-jose/v3/jwt"
+	"golang.org/x/net/context"
+	"sync"
+	"time"
+)
+
+
+var globalTrustedIdProviderManager TrustedIdProviderManagerImpl
+var globalTrustedIssuerContextTimeout = time.Second * time.Duration(5)
+
+// May be used to manage an token verification against an entire collection
+// of trusted IdP's.
+type TrustedIdProviderManager interface {
+
+	// On nil error, IDToken contains the parsed jwt token iff there exist
+	// at least one trusted identity provider in this managed set of IdP's
+	// that successfully validated this token. Else, (if not valid or not
+	// trusted or an error is encountered), a non-nil error returned and
+	// IDToken is nil.
+	VerifyToken(token *string) (*oidc.IDToken, error)
+}
+
+// Private struct to implement interface, TrustedIdProviderManager.
+// We use a sync map to safely manage a collection of trusted
+// issuers and their verifiers.
+type TrustedIdProviderManagerImpl struct {
```

Comment:
If we want to keep it as a private implementation we should rename to `trustedIdProviderManagerImpl` to avoid exporting it from the package.

I think too that you could even just name it like `oidcIDPManager` (gist being drop `impl` and instead say what implementation it is, in that example `oidcIDP` versus some like `mockTrustedIdProviderManager` or whatever else we end up with)

Reply:
Yea, originally `trustedIdProviderManagerImpl ` was actually private. Then I tried to write some tests for it, and, it became difficult, so I opted for a public struct. That comment needs to be corrected.

I like the suggestion on a simpler and more descriptive name for it, `oidcIDPs`